### ISSUE TITLE
Dependency list could be smaller

### ISSUE THREAD

**Issue by SysPete** [ironcamel/Dancer2-Plugin-Email#8]
@ironcamel I'm wondering whether the current dependency list can be trimmed:

The following are only needed for the SMTP transport and then only when auth is needed:
```
Authen::SASL   = 2.16
MIME::Base64   = 3.14
```
Also recent `Net::SMTP` has builtin support for SSL so I think these should also be removed:
```
Net::SMTP::SSL = 1.01
Net::SSLeay    = 1.58
```
Do you have a good reason why we should keep these deps?

As for `Test::WWW::Mechanize::PSGI` I'm going to switch the test to using `HTTP::Request::Common` and `Plack::Test` since D2 has those deps and we can slim down even further.

**Comment by SysPete** [comment]
Mechanize removed via 21c0cd230ac5f03019083335ac315fabf251636e. I've also switched to `Module::Runtime` via 88aa82da2d810c30032f9b72713147e3078e18a6 to reduce deps further.

**Comment by ironcamel** [comment]
I'm fine with the 2 commits removing the `Mechanize` and `Module::Runtime` deps. I would prefer to not break support for the SMTP transport however.

**Comment by SysPete** [comment]
OK, then I'll create a new release from current HEAD and have a think about the SMTP transport later.

**Comment by hvoers** [comment]
The use of Net::SMTP::SSL  is deprecated by upstream.
So please remove it.
(In the FreeBSD ports tree: EXPIRATION_DATE=2017-03-31)

**Comment by ironcamel** [comment]
We should replace `Net::SMTP::SSL` with `Net::SMTP` (based on my reading of https://metacpan.org/pod/Net::SMTP::SSL#DEPRECATED)

**Comment by SysPete** [comment]
This also means a new dep on `IO::Socket::SSL` instead of `Net::SSLeay` which I've addressed in d9d6967275df0573314fff766652c3679d0dafbb.

@hvoers thanks for the heads up.

@ironcamel if you're happy I'll cook a new version for the CPAN.

**Comment by SysPete** [comment]
v0.0202 finally on its way to CPAN. Apologies for the delay.
